src/app/clusters/icd-management-server/ICDManagementCluster.h

```diff
@@ -17,79 +17,86 @@
 
 #pragma once
 
-#include <app-common/zap-generated/cluster-objects.h>
-#include <app/AttributeAccessInterface.h>
-#include <app/CommandHandler.h>
 #include <app/icd/server/ICDConfigurationData.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/OptionalAttributeSet.h>
 #include <app/util/basic-types.h>
-#include <crypto/SessionKeystore.h>
+#include <clusters/IcdManagement/Commands.h>
+#include <clusters/IcdManagement/Metadata.h>
 #include <lib/core/Optional.h>
 #include <lib/support/Span.h>
-#include <protocols/interaction_model/StatusCode.h>
 
 #include <app/icd/server/ICDServerConfig.h>
 
 #if CHIP_CONFIG_ENABLE_ICD_CIP
-#include <app/ConcreteAttributePath.h>
-#include <app/icd/server/ICDMonitoringTable.h> // nogncheck
 #include <credentials/FabricTable.h>
+#include <crypto/SessionKeystore.h>
 #include <lib/core/CHIPPersistentStorageDelegate.h>
 #endif // CHIP_CONFIG_ENABLE_ICD_CIP
 
-namespace chip {
-namespace Crypto {
-using SymmetricKeystore = SessionKeystore;
-} // namespace Crypto
-} // namespace chip
-
 namespace chip {
 namespace app {
 namespace Clusters {
 
-class ICDManagementServer
+namespace IcdManagement {
+enum class OptionalCommands : uint32_t
```

Comment:
Could use uint8_t here for now, save a bit of RAM.  You can use the space to store the length of the trigger instruction.

Reply:
Sure